gsuite/resource_user.go

```diff
@@ -309,49 +309,6 @@ func resourceUserCreate(d *schema.ResourceData, meta interface{}) error {
 	}
 	user.SshPublicKeys = userSSHs
 
-	userPosixs := []*directory.UserPosixAccount{}
-	posixCount := d.Get("posix_accounts.#").(int)
-	for i := 0; i < posixCount; i++ {
-		posixConfig := d.Get(fmt.Sprintf("posix_accounts.%d", i)).(map[string]interface{})
-		userPosix := &directory.UserPosixAccount{}
-
-		if posixConfig["gecos"] != "" {
-			log.Printf("[DEBUG] Setting posix %d gecos: %s", i, posixConfig["gecos"].(string))
-			userPosix.Gecos = posixConfig["gecos"].(string)
-		}
-		if posixConfig["gid"] != 0 {
-			log.Printf("[DEBUG] Setting posix %d gid: %d", i, uint64(posixConfig["gid"].(int)))
-			userPosix.Gid = uint64(posixConfig["gid"].(int))
-		}
-		if posixConfig["home_directory"] != "" {
-			log.Printf("[DEBUG] Setting posix %d home_directory: %s", i, posixConfig["home_directory"].(string))
-			userPosix.HomeDirectory = posixConfig["home_directory"].(string)
-		}
-		if posixConfig["system_id"] != "" {
-			log.Printf("[DEBUG] Setting posix %d system_id: %s", i, posixConfig["system_id"].(string))
-			userPosix.SystemId = posixConfig["system_id"].(string)
-		}
-		if posixConfig["shell"] != "" {
-			log.Printf("[DEBUG] Setting posix %d shell: %s", i, posixConfig["shell"].(string))
-			userPosix.Shell = posixConfig["shell"].(string)
-		}
-		if posixConfig["primary"] != "" {
-			log.Printf("[DEBUG] Setting posix %d primary: %t", i, posixConfig["primary"].(bool))
-			userPosix.Primary = posixConfig["primary"].(bool)
-		}
-		if posixConfig["uid"] != 0 {
-			log.Printf("[DEBUG] Setting posix %d uid: %d", i, uint64(posixConfig["uid"].(int)))
-			userPosix.Uid = uint64(posixConfig["uid"].(int))
-		}
-		if posixConfig["username"] != "" {
-			log.Printf("[DEBUG] Setting posix %d username: %s", i, posixConfig["username"].(string))
-			userPosix.Username = posixConfig["username"].(string)
-		}
-
-		userPosixs = append(userPosixs, userPosix)
-	}
-	user.PosixAccounts = userPosixs
```

Comment:
I'll test this out again later, I think this might work if we first create the user without and then immediately update with this data, however all my accounts are automatically being suspended by g suite for the moment so testing that will have to follow later. Maybe tomorrow.